mono/utils/mono-threads-mach-helper.c

```diff
@@ -57,7 +57,11 @@ mono_dead_letter_dealloc (id self, SEL _cmd)
 {
 	struct objc_super super;
 	super.receiver = self;
+#if !defined(__cplusplus) && !__OBJC2__
 	super.class = nsobject;
+#else
+	super.super_class = nsobject;
+#endif
```

Comment:
FWIW, this is related: https://github.com/opensource-apple/objc4/blob/cd5e62a5597ea7a31dccef089317abb3a661c154/runtime/message.h#L37-L51